src/Voron/Data/BTrees/TreePage.cs

```diff
@@ -86,7 +86,7 @@ public ushort* KeysOffsets
 
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public TreeNodeHeader* Search(LowLevelTransaction tx, Slice key)
+        public TreeNodeHeader* Search(LowLevelTransaction tx, Slice key, bool backward = false)
```

Comment:
I have some concerns that making it an optional parameter might be error prone. It's easy to skip it when adding a new method doing backward seek in the future.

Reply:
There are 19 references to this method, I preferred to do it like this instead of changing all of the references..

Reply:
I can accept your decision. Although 19 references isn't a lot and with the suggested change we'd be on the safe side. No option someone could skip it by mistake when doing backward seek.